src/aix.rs

```diff
@@ -0,0 +1,31 @@
+use std::num::NonZeroUsize;
+use std::io::{BufReader, Read, Seek, SeekFrom};
+use std::fs::File;
+
+extern "C" {
+    fn getpid() -> i32;
+}
+
+// https://www.ibm.com/docs/en/aix/7.3?topic=files-proc-file
+//
+// The 'status' file is defined as binary format below
+//   uint32_t pr_flag;  /* process flags from proc struct p_flag */
+//   uint32_t pr_flag2; /* process flags from proc struct p_flag2 */
+//   uint32_t pr_flags; /* /proc flags */
+//   uint32_t pr_nlwp;  /* number of threads in the process */
+//   ...                /* Omitted */
+pub(crate) fn num_threads() -> Option<NonZeroUsize> {
+    let pid = unsafe {
+        getpid()
+    };
+    let file = File::open(format!("/proc/{}/status", pid)).ok()?;
+    let mut reader = BufReader::new(file);
```

Comment:
Instead of allocating a Bufreader (8k heap allocation) and then seeking (lseek() call), it would probably be more efficient to just read the first 16 bytes directly and extract from there.